modules/slugs/includes/qtranslate-slug-import.php

```diff
@@ -0,0 +1,149 @@
+ */
+function qtranxf_slugs_check_import_qts() {
+    global $wpdb;
+
+    /**
+     * Generic function that counts the slugs meta, legacy (QTS) or new (QTX).
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $prefix prefix for the meta key
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $count_slugs = function ( $table, $prefix, &$msg ) use ( $wpdb ) {
+        $results = $wpdb->get_var( "SELECT count(*) FROM  $table WHERE meta_key like '$prefix%'" );
+        if ( $results ) {
+            $msg[] = sprintf( __( "Found %s slugs from $table.", 'qtranslate' ), $results );
+        }
+    };
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_META_PREFIX, $msg );
+    if ( ! empty( $msg ) ) {
+        // Found some post/term meta with the new keys, no import to suggest (it can still be done manually).
+        return '';
+    }
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+
+    return empty ( $msg ) ? $msg : implode( '<br>', $msg );
+}
+
+/**
+ * Import slugs meta by duplicating the legacy QTS postmeta and termmeta for QTX.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_import_qts_meta( $db_commit ) {
+    global $wpdb;
+
+    $new_prefix = QTX_SLUGS_META_PREFIX;
+    $old_prefix = QTX_SLUGS_LEGACY_QTS_META_PREFIX;
+
+    /**
+     * Generic function that imports old meta into new ones. All existing new meta are erased first.
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $colid column name giving the meta id (post_id, term_id)
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $import_meta = function ( $table, $colid, &$msg ) use ( $wpdb, $old_prefix, $new_prefix ) {
+        $results = $wpdb->query( "DELETE FROM $table WHERE meta_key like '$new_prefix%'" );
+        if ( $results ) {
+            $msg[] = sprintf( __( "Deleted %s rows from $table (%s).", 'qtranslate' ), $results, $new_prefix );
+        }
+        $results = $wpdb->query( "INSERT INTO $table ($colid, meta_key, meta_value)
+                              SELECT $colid, REPLACE(meta_key, '$old_prefix', '$new_prefix'), meta_value
+                              FROM  $table
+                              WHERE meta_key like '$old_prefix%'" );
+        $msg[]   = sprintf( __( "Imported %s rows into $table (%s->%s).", 'qtranslate' ), $results ?: '0', $old_prefix, $new_prefix );
+    };
+
+    $msg = [];
+    $wpdb->query( "START TRANSACTION" );
+    $import_meta( $wpdb->postmeta, 'post_id', $msg );
+    $import_meta( $wpdb->termmeta, 'term_id', $msg );
```

Comment:
Just realized that old metadata is not removed from $wpdb->termmeta on [legacy plugin uninstallation](https://github.com/not-only-code/qtranslate-slug/blob/386b90723d79792ffe4b88138908a065e82b01f2/qtranslate-slug.php#L182).
We may take care here and remove the old ones after successful import or add a checkbox to remove the orphaned data.

Reply:
Fix needed also for current qtx uninstallation action, added in e91398359930a03d84fbf7e14fbad1a003c4d08b

Reply:
We should at least mention in import/module notes.
I submitted to original plugin repo a [PR](https://github.com/not-only-code/qtranslate-slug/pull/156) as it seems something has been merged recently.
In case it is merged we can mention that better if plugin unistallation is done from that commit on, otherwise those entries will need to be deleted manually from db.

Reply:
I see. The problem is that the current users may not have this change in their QTS plugin. Perhaps we could make a special case here and add a new option to delete explicitly the QTS meta after import (also taking the commit/rollback flag).

Reply:
I don't know what to do here. Adding a QTS cleanup is possible but it would make it too complicated maybe.

Reply:
We could add a `register_uninstall_hook` for the legacy plugin with the missing action only `$wpdb->query("DELETE from $wpdb->termmeta WHERE meta_key IN ($meta_keys)");` in

Reply:
Yes, but it should become unnecessary now with the new import-migration.
It can be added in a separate PR if needed, but I close this topic here to make the review clearer.